### DISCUSSION TITLE
Artwork Layout (.lay) File Path Support in Linux

### DISCUSSION THREAD

**Discussion by RandyR99** [mamedev/discussions#17] [Ideas]
#https://github.com/mamedev/mame/issues/10567

@cuavas

Referring to the above issue, could we add support in Linux for the import of images from a 'common' archive file? As an example, when creating artwork for chess boards, many elements are identical and space can be saved by grouping them in a separate zip file in the artworks folder. It's not an issue when only dealing with a couple of boards, but when you get up to 200 variations, those megabytes add up, and it's inefficient. It works in Windows with the current code so can we not get it working on Linux as well?

If it's just a fluke that it works in Windows, I think it's a great feature and a huge timesaver and should be supported.

Respectfully,

Randy

**Comment by cuavas** [comment]
It’s emergent behaviour on Windows, not something that’s intended to work.  It’s not supported or encouraged as it could lead to dependency hell for artwork packages.

**Comment by RandyR99** [comment]
Not for the artwork producers it wouldn't. I'm talking custom artwork, not the built-in stuff. It seems like it would be simple to implement, but I'm not a programmer.

**Comment by RandyR99** [comment]
@cuavas

Also, the built-in layouts (e.g. fidel_chesster.lay) reference the chess folder in the artwork folder. Here are a few lines from that file:

```
	<element name="piece" defstate="0">
		<image file="chess/wp.svg" state="1"/>
		<image file="chess/wn.svg" state="2"/>

```
This does not work for custom artwork in Linux so you can't even put common items in uncompressed folders and have them referenced and correctly load from the compressed layout file (e.g.chesster.zip). It doesn't make sense to me why internal artwork files allow this but not custom ones. It may not be a bug, but it's a frustrating limitation.

Randy
